packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -47,119 +42,238 @@ export function createNewConnectionInfo(): ConnectionInfo {
+        return {
+          ...state,
+          // Autoconnect is a bit of a special case where we preset the
+          // connection form with the autoconnect info so that if it fails in
+          // single connection mode, the form is pre-populated as expected
+          editingConnectionInfo: action.isAutoConnect
+            ? action.connectionInfo
+            : state.editingConnectionInfo,
+          isEditingConnectionInfoModalOpen:
+            // Close connection editing modal when connection starts
+            action.connectionInfo.id === state.editingConnectionInfo?.id
+              ? false
+              : state.isEditingConnectionInfoModalOpen,
+        };
+      case 'connection-attempt-succeeded': {
+        const connectionErrors = { ...state.connectionErrors };
+        delete connectionErrors[action.connectionInfo.id];
+
+        const oidcDeviceAuthState = { ...state.oidcDeviceAuthState };
+        delete oidcDeviceAuthState[action.connectionInfo.id];
+
+        return {
+          ...state,
+          connectionErrors,
+          oidcDeviceAuthState,
+        };
+      }
+      case 'connection-attempt-errored':
+        return {
+          ...state,
+          editingConnectionInfo:
+            // Autoconnect special case for single connection: if we failed when
+            // autoconnecting, there might be no info set in the form yet if we
+            // failed while trying to read the autoconnect info. For that case
+            // we prepopulate the editing state so that the form can map the
+            // error correctly
+            action.isAutoConnect && !state.editingConnectionInfo
+              ? action.connectionInfo
+              : state.editingConnectionInfo,
+          connectionErrors: {
+            ...state.connectionErrors,
+            [action.connectionInfo.id]: action.error,
+          },
+        };
+      case 'oidc-attempt-connect-notify-device-auth':
+        return {
+          ...state,
+          oidcDeviceAuthState: {
+            ...state.oidcDeviceAuthState,
+            [action.connectionInfo.id]: {
+              url: action.verificationUrl,
+              code: action.userCode,
+            },
+          },
+        };
+      case 'new-connection':
+        return {
+          ...state,
+          editingConnectionInfo: createNewConnectionInfo(),
+          isEditingConnectionInfoModalOpen: true,
+        };
+      case 'duplicate-connection':
+        return {
+          ...state,
+          editingConnectionInfo: action.connectionInfo,
+          isEditingConnectionInfoModalOpen: action.autoDuplicate
+            ? // When autoduplicating we just save a copy right away without
+              // opening a editing modal
+              state.isEditingConnectionInfoModalOpen
+            : true,
+        };
+      case 'edit-connection': {
+        if (
+          // In multiple connections mode we don't allow to start another edit
+          // when one is already in progress
```

Comment:
I'm not sure if this was an intentional decision, before this was not possible mostly because of the bugs in the state. Currently it should be technically possible to "open" multiple times while editing if you get a toast with an error for a different connection, but maybe it's still a good idea not to allow resetting the form while one is already opened